tool/build/apelink.c

```diff
@@ -979,13 +988,24 @@ static void AddLoader(const char *path) {
   if (loaders.n == ARRAYLEN(loaders.p)) {
     Die(prog, "too many loaders");
   }
-  loaders.p[loaders.n++].path = path;
+  struct Loader *loader = &loaders.p[loaders.n++];
+  loader->path = path;
+  if (loader_kernel[0] != '\0') {
+    strncpy(loader->kernel, loader_kernel, sizeof(loader->kernel));
+  } else {
+    loader->kernel[0] = '\0';
+  }
+}
+
+static void SetLoaderKernel(const char *kernel) {
+  strncpy(loader_kernel, kernel, sizeof(loader_kernel));
+  loader_kernel[sizeof(loader_kernel) - 1] = 0;
```

Comment:
Just use strlcpy(). In fact, technically speaking, it would be valid to have the static field be a `const char *` pointer, since the memory pointed to by `optarg` doesn't change. Then the code above could be updated to simply copy the pointer over. Then later, when you do your strcmp() it'll actually go very faster since the pointers will be equal in practice.

https://github.com/jart/cosmopolitan/blob/b235492e715df777bd14424dc1b7f9cd0605b379/libc/intrin/strcmp.c#L31-L36